Tests/scripts/lock_cloud_machines.py

```diff
@@ -172,81 +201,158 @@ def try_to_lock_machine(storage_bucket: any, machine: str, machines_locks: list,
     lock_machine_name = ''
     job_id_of_the_existing_lock = next((d['job_id'] for d in machines_locks if d["machine_name"] == machine), None)
     if job_id_of_the_existing_lock:
-        logging.info(f'There is a lock file for job id: {job_id_of_the_existing_lock}')
+        logging.debug(f'There is a lock file for job id: {job_id_of_the_existing_lock}')
         job_id_of_the_existing_lock_status = check_job_status(gitlab_status_token, job_id_of_the_existing_lock)
-        logging.info(f'the status of job id: {job_id_of_the_existing_lock} is: {job_id_of_the_existing_lock_status}')
+        logging.debug(f'the status of job id: {job_id_of_the_existing_lock} is: {job_id_of_the_existing_lock_status}')
         if job_id_of_the_existing_lock_status != 'running':
             # machine found! removing the not relevant lock and create a now one
             remove_machine_lock_file(storage_bucket, gcs_locks_path, machine, job_id_of_the_existing_lock)
-            lock_machine(storage_bucket, gcs_locks_path, machine, job_id)
-            lock_machine_name = machine
+        else:
+            return lock_machine_name
     else:
         # machine found! create lock file
-        logging.info('There is no a lock file')
-        lock_machine(storage_bucket, gcs_locks_path, machine, job_id)
-        lock_machine_name = machine
+        logging.info('There is no existing lock file')
+    logging.info(f'Locking machine {machine}')
+    lock_machine(storage_bucket, gcs_locks_path, machine, job_id)
+    lock_machine_name = machine
     return lock_machine_name
 
 
-def main():
-    install_logging('lock_cloud_machines.log', logger=logging)
-    logging.info('Starting search a CLOUD machine to lock')
-    options = options_handler()
-    storage_client = storage.Client.from_service_account_json(options.service_account)
-    storage_bucket = storage_client.bucket('xsoar-ci-artifacts')
-    number_machines_to_lock = int(options.number_machines_to_lock)
+def get_and_lock_all_needed_machines(storage_client, storage_bucket, list_machines, gcs_locks_path,
+                                     number_machines_to_lock, job_id, gitlab_status_token):
+    """
+    get the requested machines and locked them to the job-id.
```

Comment:
I think we should elaborate a bit more when it comes to more sophisticated functions
```suggestion
    get the requested machines and locked them to the job-id. The function will wait (busy waiting) until it was able to successfully lock the requested number of machines. In between runs, it will sleep for a minute to allow other builds to finish.
```